torch/csrc/jit/graph_executor.cpp

```diff
@@ -215,7 +219,7 @@ struct GraphExecutorImpl {
   : GraphExecutorImpl(graph, optimize, isDifferentiable(*graph)) {}
 
   // entry point where execution begins
-  variable_tensor_list run(variable_tensor_list inputs) {
+  void run(Stack & inputs) {
```

Comment:
In all other uses of stacks, the assumption is that the stack can hold more inputs that what the operator will consume. Here it is adding a strong assumption that the stack only contains the inputs for this executor. It would be safer if we only examined the top N elements as determined by the graph.

Reply:
I'm pretty sure that the executor would just completely ignore all extra stack contents, but I can explicitly pop the appropriate number of inputs into a new stack (this will be less efficient though). I've been recently wondering why do we call this thing a stack. Do we ever have any situations where an op is called, but it shouldn't consume all values on the stack (i.e. does it ever contain certain `IValue`s for longer than a single op call)?